### PR TITLE
refactor: resolve all circular dependency exceptions

### PR DESCRIPTION
**Pull request by alxhub**
The circular deps test protects against real, runtime circular dependencies but will pick up false positives when a cycle only exists at the type level. Angular has a number of such type-only cycles, which were listed as exceptions in a circular deps golden file.

Having these exceptions is undesirable, because they can prevent *real* circular dependencies from being caught by the test. This commit clears all current exceptions by leveraging a recent change to circular dep testing: the test now understands `import type` syntax and does not consider such imports as contributing to import cycles.

`import type` is used to mark imports that were involved in prior cycles in a way that satisfies the circular dep test. All existing entries in the golden file are resolved via this technique.

### CODE REVIEW

**Review by AndrewKushnir — APPROVED** [review]
Yay! 🎉

**Comment by JeanMeche** [comment]
@crisbeto  did a test for this in #50038 and mentionned [here](https://github.com/angular/angular/pull/50051#issuecomment-1526496436) some technical limitations. Have they been fixed ? What were they ?

File: goldens/circular-deps/packages.json [hunk 1]
```diff
@@ -1,314 +1 @@
...
-    "packages/router/src/navigation_transition.ts",
-    "packages/router/src/operators/resolve_data.ts"
-  ],
-  [
-    "packages/router/src/shared.ts",
-    "packages/router/src/url_tree.ts"
-  ]
-]
\ No newline at end of file
+[]
```
Comment (JeanMeche) [inline goldens/circular-deps/packages.json:1]: I just ran `yarn -s ng-dev ts-circular-deps approve --config ./packages/circular-deps-test.conf.js` is this expected  ?

**Comment by alxhub** [comment]
(easier to regenerate)

**Pull request closed without merging**